podcast/2.15.md

```diff
@@ -0,0 +1,1299 @@
+
+**Beyang Liu:**
+
+Interesting. And it was completely automated?
+
+**Sugu Sougoumarane:**
+
+Yeah, so it would run every night, look at all the transactions, and look at every user's balance. And it had to be done efficiently, because you can't loop through users. So I had to run some smart queries that did diffs in different formats. And then that was used to identify the offending transaction. And so, people loved me for it. They said, "Wow. Sugu solved a real problem."
+
+**Beyang Liu:**
+
+Sounds like you're a hero. I mean, every single day when an engineer would be on call to handle these things.
+
+**Sugu Sougoumarane:**
+
+Otherwise, I was not one of the elite engineers. So after that, they said, "No, Sugu is cool. If there's a problem, we can count on Sugu to solve it."
+
+**Beyang Liu:**
+
+That's fascinating. So they respected you.
+
+**Sugu Sougoumarane:**
+
+So I kind of earned the respect. It was very hard to earn that type of respect. You had to show something for it.
+
+**Beyang Liu:**
+
+So how did all this relate eventually to the work on Vitess?
+
+**Sugu Sougoumarane:**
+
+After that, when they said “Sugu solves problems,” I kind of went into solving... So, at that point, the biggest challenge for PayPal was always scaling. I became part of that core team that was responsible for scaling PayPal. And this is another story you may not know, we tried to shard PayPal.
+
+**Beyang Liu:**
+
+You tried to shard it? Like split the databases into pieces?
+
+**Sugu Sougoumarane:**
+
+Split the database into smaller parts. And that experiment failed, not for technical reasons, but nevertheless, the experiment failed. And we had to rollback.
+
+**Beyang Liu:**
+
+So all PayPal was literally on a single giant database.
+
+**Sugu Sougoumarane:**
+
+Yes.
+
+**Beyang Liu:**
+
+What was the database, just out of curiosity?
+
+**Sugu Sougoumarane:**
+
+Oracle.
+
+**Beyang Liu:**
+
+Oh, Oracle. Okay. Oracle. So one giant Oracle database instance.
+
+**Sugu Sougoumarane:**
+
+It was Oracle, and we had the biggest hardware money could buy.
+
+**Beyang Liu:**
+
+Which was what in those days? Do you remember?
+
+**Sugu Sougoumarane:**
+
+It was called the Sun E15K, I believe. It was the biggest machine Sun ever produced. If Sun made a new, more powerful machine, we bought it. So I think we had an E10K before that. And then after that we bought something called the E15K.
+
+**Beyang Liu:**
+
+It was one box? Literally one box.
+
+**Sugu Sougoumarane:**
+
+One box. We had two E15Ks because there was a hot standby.
+
+**Beyang Liu:**
+
+Okay. Got it. Yeah, for fallover.
+
+**Sugu Sougoumarane:**
+
+So that's how Oracle ran.
+
+**Beyang Liu:**
+
+Makes sense.
+
+**Sugu Sougoumarane:**
+
+The only hint I can give you is, after that outage we migrated out of E15K into Fujitsu. That happened after I left, but there is a hint in there.
```

Comment:
Is there a prior mention of the outage referenced here? It seems to come out of the blue